spring-integration-core/src/main/java/org/springframework/integration/endpoint/MessageProducerSupport.java

```diff
@@ -182,6 +200,13 @@ protected void onInit() {
 	 */
 	@Override
 	protected void doStart() {
+		if (this.reactive && this.flux == null) {
+			this.flux =
+					Flux.<Message<?>>create(emitter -> this.sink = emitter, FluxSink.OverflowStrategy.IGNORE)
```

Comment:
it is not recommended to leak `emitter`. Use one of the Processors:
https://projectreactor.io/docs/core/release/reference/#processors

Reply:
Actually we've been advised exactly opposite: the processors are going to go away from Reactor Core, so would be better to rely on the similar structure provided by the `Flux.create()`.

Anyway this PR is too old and really deserves revisiting one day.

I hope you'll come up soon with the `Flux.fromQueue()` as we discussed recently in call, so we would have a natural back-pressure for the constantly emitting source.

Thank you for feedback, @bsideup , anyway!

Reply:
They are not going away, this is a misunderstanding. `Flux.fromQueue` won't give you backpressure I think, the processes is the way to go :)

Reply:
Yeah... Let's reconsider that in the next milestone!
We have a lot of similar `Flux.create()` here in Spring Integration.

I'll ping you later if needed.

Thank you, Sergey, again!

Reply:
np! Feel free to assign me as a reviewer, or ask for more info directly 👍